var/spack/repos/builtin/packages/gcc/package.py

```diff
@@ -973,29 +1206,62 @@ def write_rpath_specs(self):
 
         # Find which directories have shared libraries
         rpath_libdirs = []
-        for dir in ["lib", "lib64"]:
-            libdir = join_path(self.prefix, dir)
-            if glob.glob(join_path(libdir, "*." + dso_suffix)):
-                rpath_libdirs.append(libdir)
+        prefixes = [self.prefix]
+        if '+stage1' in self.spec:
+            prefixes.append(self.spec['glibc'].prefix)
+        for pfx in prefixes:
+            for dir in ["lib", "lib64"]:
+                libdir = join_path(pfx, dir)
+                if glob.glob(join_path(libdir, "*." + dso_suffix)):
+                    rpath_libdirs.append(libdir)
 
         if not rpath_libdirs:
             # No shared libraries
             tty.warn("No dynamic libraries found in lib/lib64")
             return
 
-        # Overwrite the specs file
+        add_sysroot = self.spec.satisfies("os=spack")
         with open(specs_file, "w") as out:
+            loader = re.compile(':-dynamic-linker ')
+            next = False
+            name = None
             for line in lines:
+                if add_sysroot:
+                    if loader.search(line):
```

Comment:
Instead of patching this post-hoc, you can also iirc modify a config file. That should have the advantage that if you're doing multistage stuff, you can influence gcc in those stages (that may not be relevant here though...)

Reply:
Oh that would be nice actually, do you have a ref? I couldn’t find where they were doing it for the spec, only saw them wrapping xgcc and using their scripts. We’re already patching configure and a couple others and that might simplify it all.